galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerVolumeImpl.java

```diff
@@ -0,0 +1,115 @@
+package dev.galasa.docker.internal;
+
+import com.google.gson.JsonObject;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import dev.galasa.docker.DockerManagerException;
+import dev.galasa.docker.IDockerVolume;
+
+/**
+ * A implementation of the docker volumes that will be brought up on the engine
+ * 
+ * @author James Davies
+ */
+public class DockerVolumeImpl implements IDockerVolume {
+    private final String volumeName;
+    private final String mountPath;
+    private final DockerEngineImpl engine;
+    private final boolean readOnly;
+
+    private static final Log logger = LogFactory.getLog(DockerVolumeImpl.class);
+
+    /**
+     * Constructor that determines the nature of the volume (readOnly or not), and provisions or ensures
+     * the volumes exsists.
+     * 
+     * @param volumeName
+     * @param mountPath
+     * @param engine
+     * @throws DockerManagerException
+     */
+    public DockerVolumeImpl(String volumeName, String mountPath, DockerEngineImpl engine)
+            throws DockerManagerException {
+
+        // Non Galasa made volumes should not be edited by Galasa run containers.
+        if ("".equals(volumeName)) {
+            this.readOnly = false;
+            logger.info("Generating volume");
+
+            // If no named passed, Docker will name the volume. We will add galasa labels.
+            JsonObject json = engine.createVolume(volumeName);
+            this.volumeName = json.get("Name").getAsString();
+        } else {
+            this.volumeName = volumeName;
```

Comment:
The volume name must be prefixed with the RunName.
Therefore, I do not believe the readonly is required.